cli/internal/globby/globby.go

```diff
@@ -1,14 +1,28 @@
 package globby
 
 import (
-	"github.com/vercel/turborepo/cli/internal/fs"
-
+	"os"
 	"path/filepath"
 	"strings"
 
 	"github.com/bmatcuk/doublestar/v4"
+	"github.com/spf13/afero"
 )
 
+var afs *afero.Afero
+
+func init() {
+	setFileSystem(afero.NewOsFs())
```

Comment:
I don't know what the Go idiomatic approach for redefining a package variable at testing time would be, so the implementation I chose may be nonsense. Is there a better way to do this?

Reply:
FWIW, there is a [preference](https://github.com/uber-go/guide/blob/master/style.md#avoid-init) (but not hard rule) to avoid `init` functions.

I think the way I would handle this is to do something along the lines of:
```go
var osFS := afero.NewOs()

func GlobFiles(...) []string {
  return GlobFilesWithFS(osFS, ...)
}

func GlobFilesWithFS(fs afero.Afero, ...) []string {
  // implementation
}
```

That way tests can use `GlobFilesWithFS` and existing calling code doesn't have to change. Then once we've hoisted an instance of `afero.Afero` up to a field on `Config` or equivalent, we can delete `GlobFiles`, rename `GlobFilesWithFS -> GlobFiles` and update the calling locations.

Reply:
I saw that pattern throughout `afero` but I wasn't convinced that it was elegant. Go function argument pattern matching when? 😜 (Developing "taste" for Go is gonna be a ride.)

I've switched to just passing in the FS and also removed the receiver. Our library code will be a bit more verbose (e.g. `afero.Thing(fs, ...)`) and we can go back to having a receiver once we have a singleton hoisted to `fs`.

Reply:
(I really appreciate the link to the style guide, I'll read through the entire thing this weekend.)